### PR TITLE
DEV 2097 drop py36 

### PR DESCRIPTION
**Pull request by qiaouchicago**
- [x] drop py36 in ci
- [x] pyupgrade

* drop py36 in piepelines; add templates/common/python.yaml in gitlab-ci

* update python_versions in gitlab-ci, update .tox_script, update setup.py

* pyupgrade --py37-plus src/**/*py tests/*py

* add MANIFEST.in

* remove [dev-]requirements.txt

* overwrite tox parallel

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -14,7 +14,22 @@ include:
     file:
       - templates/global/full.yaml
       - templates/python/full.yaml
+      - templates/common/python.yaml
 
+.python_versions:
+  parallel:
+    matrix:
+      - BUILD_PY_VERSION: [python3.7]
```
Comment (stilesj-uchicago) [inline .gitlab-ci.yml:22]: Shouldn't' there be multiple python versions specified here? One for each supported version?

**Review by stilesj-uchicago — APPROVED** [review]

**Review by wrkhenddher — APPROVED** [review]
I suppose there's a ticket to add black pre-commit (and others). I'm guessing they're not there yet because of the single-quoted strings.

File: .gitlab-ci.yml (hunk 1 above)
Comment (qiaouchicago) [reply .gitlab-ci.yml]: added 3.8 and 3.9

**Comment by qiaouchicago** [comment]
> I suppose there's a ticket to add black pre-commit (and others). I'm guessing they're not there yet because of the single-quoted strings.

The reason I didn't add pre-commit is, there are lots of yaml files, they are used by scripts of bio team. 
@philtom-ctds  concerns reformatting those yaml might cause compatibility issues for bio.

**Pull request merged**